kubeadm/scripts/PrepareNode.ps1

```diff
@@ -73,7 +74,7 @@ if ($netId.Length -lt 1) {
     docker network create -d nat host
 }
 
-$cmd = "C:\k\kubelet.exe $global:KubeletArgs --cert-dir=$env:SYSTEMDRIVE\var\lib\kubelet\pki --config=/var/lib/kubelet/config.yaml --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --hostname-override=$(hostname) --pod-infra-container-image=`"mcr.microsoft.com/oss/kubernetes/pause:1.3.0`" --enable-debugging-handlers --cgroups-per-qos=false --enforce-node-allocatable=`"`" --network-plugin=cni --resolv-conf=`"`" --log-dir=/var/log/kubelet --logtostderr=false --image-pull-progress-deadline=20m"
+$cmd = "C:\k\kubelet.exe $global:KubeletArgs --cert-dir=$env:SYSTEMDRIVE\var\lib\kubelet\pki --config=/var/lib/kubelet/config.yaml --bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf --hostname-override=$(hostname) --pod-infra-container-image=`"mcr.microsoft.com/oss/kubernetes/pause:1.3.0`" --enable-debugging-handlers --cgroups-per-qos=false --enforce-node-allocatable=`"`" --network-plugin=cni --resolv-conf=`"`" --log-dir=/var/log/kubelet --logtostderr=false --image-pull-progress-deadline=20m --node-ip=$env:NODE_IP"
```

Comment:
hi, just to double check. PowerShell will return "" here in case the user has not set the `NODE_IP` env. var, correct?

Reply:
Yeah, it will. What does `--node-ip=""` do?

Reply:
tells the kubelet to pick an IP from one of the interfaces on the node. on Linux at least, this usually ends up being the external one.

it's only documented with source code comments.

Reply:
From my test Powershell will return NULL if `NODE_IP` is not set. But I think kubelet can still startup with param `--node-ip=`.

Reply:
does it return "NULL" as a string?
if you pass "NULL" to the kubelet it would fail; it must be "".

Reply:
> does it return "NULL" as a string?
> if you pass "NULL" to the kubelet it would fail; it must be "".

Nope, it just retun a NULL value, neither "NULL" in string or empty string. The param will be `--node-ip=`.

Reply:
so if it's `--node-ip=` it will break...

i propose to add a PowerShell `if` check above the kubelet call to optionally add the whole `--node-ip=value` only if the envvar NODE_IP is set.

Reply:
pseudo code:
```
var nodeIPFlag = ""
if env(NODE_IP) != "" {
   nodeIPFlag="--node_ip="+env(NODE_IP)
}
kubelet .... $nodeIPFlag
```

Reply:
@neolit123 , It seems that `kubelet` will retrive params from file `kubeadm-flags.env`. So I think user can specify the `--node-ip` in this file for the NodeIP config.

By this way we don't need to change the `PrepareNode.ps1` or `StartKubelet.ps1` for this config.

I will close the PR and Issue. Thanks @neolit123  and @benmoss a lot for your review.

```
$FileContent = Get-Content -Path "/var/lib/kubelet/kubeadm-flags.env"
$global:KubeletArgs = $FileContent.Trim("KUBELET_KUBEADM_ARGS=`"")
...
$cmd = "C:\k\kubelet.exe $global:KubeletArgs...
```
Here's the content of `kubeadm-flags.env`
```
cat /var/lib/kubelet/kubeadm-flags.env
KUBELET_KUBEADM_ARGS="--cgroup-driver= --network-plugin=cni --pod-infra-container-image=k8s.gcr.io/pause:3.1"
```